dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
So in fact the documentation / specification of dart atomic operations is missing. I'm fine with it if dart works only on integral types or types of a given size (and this is documented!), however this is not a reason to restrict the functionality on the DASH side. DASH is intended to follow the STL as close as possible.

I see it similar to using `std::atomic` on CPUs which do not support lock-free atomic instructions for a given type: `std::atomic` still guarantees the semantics, but is implemented differently. I propose to add a trait to check if the type is natively supported by DART atomic. If so, the `GlobRef<Atomic<T>>` uses the dart atomic functions, if not it uses locks.

Of course, some functions are only available for integral types, but that is another topic.